Automattic/jetpack
# VideoPress block: Fix Cancel button and provide better error message

This PR adds a better error message when there's an error not related to the upload itself.

![Captura de tela de 2022-06-15 10-33-15](https://user-images.githubusercontent.com/971483/173895872-0c41e7da-40db-4a3a-bf39-7b93e10f8b88.png)

The image above shows the generic error message we see in all cases.

For example, when a user uploads a video in a format that is not supported the error message is not only incomplete but also misleading, as it recommends the user to check their connection. This error only happens when the video upload is successful, so there's nothing wrong with the connection or the upload process.

I believe the case above is pretty common.

Another situation is when a user of the Free plan tries to upload their second video. They will see this generic and misleading error message. This PR fixes it.

This PR also fixes a bug that was preventing the "Cancel" button of the error component from working.

#### Changes proposed in this Pull Request:

* Fix Cancel button on VideoPress block and provided better error message when video foramt is not supported

#### Jetpack product discussion


NA

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






Activate VideoPress on a site running the Free plan.

Confirm the error before checking out this branch:

* Try to upload a unsupported video format (for example, .mkv).
* Confirm the error message says "An error was encountered during the upload. Check your network connection."
* Click the "Cancel" button and confirm it does not work and throws error in the console

Confirm the fix by checking out this branch and building Jetpack:

* Try to upload a unsupported video format (for example, .mkv).
* Confirm you see a specific error message about the format of the video, with a link to the support docs

![Captura de tela de 2022-06-17 10-31-12](https://user-images.githubusercontent.com/971483/174308548-4b85cc7c-9226-428d-b04b-7f852ac50a70.png)


* Click Cancel and confirm it works, by resetting the Video block
* Now upload a valid video and confirm it gets added to the post
* Try to upload a second video and confirm you see an error message suggesting you to upgrade:

![Captura de tela de 2022-06-15 15-04-05](https://user-images.githubusercontent.com/971483/173896221-3b5552a6-5dd0-4452-bdd4-4c3253ee4775.png)
